<?xml version="1.0"?>
<launch>
    <arg name="root_link" default="base_link" />
    <arg name="tip_link" default="tool0" />

    <!-- Load the robot -->
    <include file="$(find ur_description)/launch/load_ur3e.launch" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

    <!-- Octomap Server Setup -->
    <node name="create_cloud_xyzrgb" pkg="octomap_filter" type="create_cloud_xyzrgb.py" />

    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server" args="">
        <param name="resolution" value="0.05" />
        <param name="frame_id" type="string" value="map" />
		<param name="sensor_model/max_range" value="4.0" />
        <remap from="cloud_in" to="artificial_cloud" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="map_to_world_broadcaster" args="0 0 0 0 0 0 map world 100" />
    <node pkg="tf" type="static_transform_publisher" name="world_to_base_broadcaster" args="0 0 0 0 0 0 world base_link 100" />

    <!-- IK Server Setup -->
    <node name="constrained_manipulability_server" pkg="constrained_manipulability" type="constrained_manipulability_server" respawn="false" output="screen" clear_params="true" launch-prefix="">
        <param name="root" type="string" value="$(arg root_link)" />
        <param name="tip" type="string" value="$(arg tip_link)" />
        <param name="robot_desc" type="string" value="robot_description" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find constrained_manipulability)/rviz/octomap_robot.rviz" />
</launch>
